lesson15_homework_sims/homework15.py

```diff
@@ -0,0 +1,131 @@
+from random import randint
+
+
+class NotEnoughMoney(Exception):
+    def __init__(self, some_to_buy):
+        self.some_to_buy = some_to_buy
+
+    def __str__(self):
+        return f"Not enough money to buy {self.some_to_buy}"
+
+
+class ChattelFirst(Exception):
+    def __str__(self):
+        return "You must buy chattel before buy estate"
+
+
+class Property:
+    def __init__(self, price, name):
+        self.__price = price
+        self.__name = name
+
+    @property
+    def price(self):
+        return self.__price
+
+    def __repr__(self):
+        return f"{self.__name} - {self.price}"
+
+
+class Chattel(Property):
+    @property
+    def type_of_property(self):
+        return 'chattel'
+
+
+class Estate(Property):
+    @property
+    def type_of_property(self):
+        return 'estate'
+
+
+class Human:
+    def __init__(self, name):
+        self.name = name
+        self.money = 0
+        self.property = []
+
+    def __repr__(self):
+        return f"{self.name} - {self.money}, {self.property}"
+
+    def is_chattel_in_property(self):
+        for property in self.property:
+            if property.type_of_property == 'chattel':
+                return True
+
+    def work(self):
+        self.money += randint(5, 10)
+
+    def buy(self, some_property):
+        if some_property.type_of_property == 'estate':
+            if self.is_chattel_in_property():
+                if some_property.price > self.money:
+                    print(f'Sorry, you do not have enough money to buy {some_property}')
```

Comment:
hmm.. why are u not using your custom exception here?

Reply:
because when I use exception, the program stops completely, and I wanted to show how Human tried to buy it failed and he went to work further